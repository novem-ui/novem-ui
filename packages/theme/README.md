# `@novem-ui/theme`

> TODO: description

## Usage

```
const theme = require('@novem-ui/theme');

// TODO: DEMONSTRATE API
```
