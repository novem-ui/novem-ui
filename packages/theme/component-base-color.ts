export { ComponentBaseColorProps, ComponentBaseColorProvider, useComponentBaseColor } from './src/component-base-color'
