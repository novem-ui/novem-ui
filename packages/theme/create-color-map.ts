export { default } from './src/utils/create-color-map'
