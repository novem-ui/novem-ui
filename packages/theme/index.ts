export { default as useTheme } from './use-theme'
export { default as withTheme } from './with-theme'
export { default as createColorMap } from './create-color-map'
export { default as ThemeProvider } from './theme-provider'
export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue, Shadows } from './theme'
