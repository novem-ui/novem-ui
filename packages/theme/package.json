{
  "name": "@novem-ui/theme",
  "version": "0.0.2-alpha.0",
  "description": "Theme for novem-ui ",
  "author": "Leonel Vieyra <me@leonelv.com>",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/theme#readme",
  "license": "ISC",
  "main": "index.js",
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/theme"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/theme --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "polished": "^4.1.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
