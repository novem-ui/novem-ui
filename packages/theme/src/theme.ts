import colors from './colors'
import * as typography from './typography'
import spacing from './spacing'
import { WeightedColorKey } from '../types'

const theme = {
  colors,
  typography,
  spacing,
}

export type BaseColor = `${keyof typeof colors.palette}`
export type Color = `${keyof typeof colors.palette}.${WeightedColorKey}`
export type HeadingSizeValue = keyof typeof typography.headingFontSizes
export type SpacingValue = keyof typeof spacing
export type TextSizeValue = keyof typeof typography.textFontSizes
export type Theme = typeof theme


export default theme
