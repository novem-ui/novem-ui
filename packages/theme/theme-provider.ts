export { default } from './src/theme-provider'
