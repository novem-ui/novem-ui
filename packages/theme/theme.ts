export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue } from './src/theme'
