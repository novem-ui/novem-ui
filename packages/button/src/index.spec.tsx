import React from 'react'

import { withTheme } from '@novem-ui/theme'
import { render } from '@testing-library/react'

import Button from '.'

describe('<Button />', () => {
  test('should match snapshot (default)', () => {
    const Component = () => <Button>Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (medium)', () => {
    const Component = () => <Button size="md">Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (small)', () => {
    const Component = () => <Button size="sm">Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (large)', () => {
    const Component = () => <Button size="lg">Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (quiet)', () => {
    const Component = () => <Button hierarchy="quiet">Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (alternative color)', () => {
    const Component = () => <Button baseColor="green">Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (alternative color)', () => {
    const Component = () => <Button disabled>Dummy text</Button>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })
})
