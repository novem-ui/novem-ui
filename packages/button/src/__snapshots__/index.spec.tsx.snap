// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot (alternative color) 1`] = `
.emotion-0 {
  background-color: #59BC6F;
  color: rgba(255,255,255,1);
  border-radius: 3rem;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  padding: 0.375rem 1rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem #59BC6F;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
  >
    Dummy text
  </button>
</div>
`;

exports[`<Button /> should match snapshot (alternative color) 2`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 3rem;
  border: none;
  cursor: auto;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  padding: 0.375rem 1rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem #F1694B;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
    disabled=""
  >
    Dummy text
  </button>
</div>
`;

exports[`<Button /> should match snapshot (default) 1`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 3rem;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  padding: 0.375rem 1rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem #F1694B;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
  >
    Dummy text
  </button>
</div>
`;

exports[`<Button /> should match snapshot (large) 1`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 3rem;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  font-weight: 600;
  padding: 0.875rem 1.5rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem #F1694B;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
  >
    Dummy text
  </button>
</div>
`;

exports[`<Button /> should match snapshot (medium) 1`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 3rem;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  padding: 0.375rem 1rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem #F1694B;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
  >
    Dummy text
  </button>
</div>
`;

exports[`<Button /> should match snapshot (quiet) 1`] = `
.emotion-0 {
  background-color: rgba(241,105,75,0.15);
  color: #F1694B;
  border-radius: 3rem;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  padding: 0.375rem 1rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem rgba(241,105,75,0.15);
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
  >
    Dummy text
  </button>
</div>
`;

exports[`<Button /> should match snapshot (small) 1`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 3rem;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  padding: 0.28rem 0.75rem;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:hover {
  box-shadow: 0 0 0.25rem #F1694B;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.emotion-0:disabled {
  background-color: rgba(31, 26, 24, 0.08);
}

.emotion-0:disabled:hover {
  box-shadow: 0 0 0.25rem rgba(31, 26, 24, 0.08);
}

<div>
  <button
    class="emotion-0"
  >
    Dummy text
  </button>
</div>
`;
